### ISSUE TITLE
`needless_slice`

### ISSUE THREAD

**Issue by JonathanWoollett-Light** [rust-lang/rust-clippy#12518]
### What it does

Checks for statically sized slices that can be constant arrays.

### Advantage

- Supports better compile-time checking and optimization.

### Drawbacks

- Requires usage of `&` prefix or `.as_slice()` suffix.

### Example

```rust
static MY_SLICE: &[u8] = &[1, 2, 3, 4, 5, 6];
my_function_that_takes_a_slice(MY_SLICE);
```

Could be written as:

```rust
const MY_ARRAY: [u8; 6] = [1, 2, 3, 4, 5, 6];
my_function_that_takes_a_slice(&MY_ARRAY);
```

**Comment by Centri3** [comment]
There's basically no difference whatsoever, anything that could be done at compile-time is still done with a slice in a constant context - the length is known at compile time, through `.len()`, so why wouldn't it be? Also, there's no assembly difference in every case I've tried, LLVM (and our own MIR passes) are really good at optimizing this :)

Really, the actual concern is converting back to an array; using `.try_into().unwrap()`. It's a code smell. Even then, using a slice instead is often done to omit specifying the length, since `[T; _]` is unstable. So this should probably be pedantic since this is really common

If we can isolate where converting back to an array is common, this should definitely be warn-by-default, but that'd require global analysis and might be out of scope of clippy.
